lib/ace/keyboard/textinput.js

```diff
@@ -300,7 +298,12 @@ var TextInput = function(parentNode, host) {
             }
             restoreStart -= i-1;
             restoreEnd -= i-1;
-            inserted = inserted.slice(0, inserted.length - i+1);
+            var endIndex = inserted.length - i + 1;
```

Comment:
I'm not sure what all the index logic in the sendText method is doing. What is the problem it's trying to solve? :)

Reply:
when character is removed in a string consisting of the same characters, endIndex becomes negative, causing editor to duplicate the text instead of removing it.